### PR TITLE
Website Redesign

### PR DESCRIPTION
**Pull request by fmaclen**
**Changes:**
- Re-arranged navigation
- Added sitemap in the footer
- Adjusted font sizes to improve hierarchies and readability
- Added anchor tags to improve navigation within pages (specially the **Plugins** section)
- Replaced **Press** with a **Links** section
- Fixed broken links
- Refreshed project's branding with a new logo
- Added media queries to improve access on mobile devices
- Used [BEM syntax](http://getbem.com/) for CSS selectors.

Did my best to stick to the existing indentation.

Additional discussion: https://github.com/jeremyevans/sequel/discussions/1863

### CODE REVIEW

File: www/layout.html.erb [hunk 1]
```diff
@@ -1,47 +1,99 @@
 <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
   "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
 
-<html xmlns="http://www.w3.org/1999/xhtml">
+<html xmlns="http://www.w3.org/1999/xhtml" class="html">
 <head>
-  <title>Sequel: The Database Toolkit for Ruby<%= " - #{title.capitalize}" unless title == 'index' %></title>
+  <title><%= "#{title.capitalize} — " unless title == 'index' %>Sequel: The Database Toolkit for Ruby</title>
```
Comment (fmaclen) [inline www/layout.html.erb:6]: Reversed the order of the `<title>` so the section reads first.

File: www/pages/index.html.erb [hunk 2]
```diff
@@ -1,40 +1,47 @@
...
+</nav>
+</div>
 
-<pre><code class='ruby' lang='ruby'><span class='ident'>require</span> <span class='punct'>&quot;</span><span class='string'>sequel</span><span class='punct'>&quot;</span>
+<ul class="index__ul">
+<li class="index__li">Thread safety, connection pooling and a concise DSL for constructing SQL queries and table schemas.</li>
+<li class="index__li">Comprehensive ORM layer for mapping records to Ruby objects and handling associated records.</li>
+<li class="index__li">Advanced database features such as prepared statements, bound variables, stored procedures, savepoints, two-phase commit, transaction isolation, primary/replica configurations, and database sharding.</li>
+<li class="index__li">With adapters for ADO, Amalgalite, IBM_DB, JDBC, MySQL, Mysql2, ODBC, Oracle, PostgreSQL, SQLAnywhere, SQLite3, and TinyTDS.</li>
+</ul>
```
Comment (fmaclen) [inline www/pages/index.html.erb:19]: Removed the _Sequel..._ at the beginning of each sentence.

File: www/pages/index.html.erb [hunk 3]
```diff
@@ -1,40 +1,47 @@
...
 
-<span class='comment'># connect to an in-memory database</span>
-<span class='constant'>DB</span> <span class='punct'>=</span> <span class='constant'>Sequel</span><span class='punct'>.</span><span class='ident'>sqlite</span>
+<a name="example"></a>
+<pre class="pre"><code class="ruby" lang="ruby"><span class="ident">require</span> <span class="punct">&quot;</span><span class="string">sequel</span><span class="punct">&quot;</span>
```
Comment (fmaclen) [inline www/pages/index.html.erb:22]: Converted all single quotes to double quotes so it's consistent with the other HTML files.

File: www/pages/links.html.erb [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+          July 2007
+        </li>
+      </ul>
+    </div>
+    <div class="row__item">
+      <h3 class="h3">Print</h3>
+      <ul class="ul">
+        <li class="ul__li">
+          <a  class="a" href="https://web.archive.org/web/20110209051845/http://www.impressjapan.jp/books/2579">Ruby! Vol. 1</a>
+          In japanese (thanks Shunichi Arai)
```
Comment (fmaclen) [inline www/pages/links.html.erb:40]: Update link to the archived version.

File: www/pages/links.html.erb [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+      <ul class="ul">
+        <li class="ul__li"><strong>RubyDay 2020</strong> <a class="a" href="https://speakerdeck.com/janko_m/sequel-when-activerecord-is-not-enough">Sequel: When Active Record is not enough</a></li>
+        <li class="ul__li"><strong>RubyC 2014</strong> <a class="a" href="http://code.jeremyevans.net/presentations/rubyc2014-2/index.html">Give-and-Go with PostgreSQL and Sequel</a> <a class="a" href="http://www.youtube.com/watch?v=toAcnwqlU1Q">Video</a></li>
+        <li class="ul__li"><strong>RubyC 2014</strong> <a class="a" href="http://code.jeremyevans.net/presentations/rubyc2014-1/index.html">Deep Dive Into Eager Loading Limited Associations</a> <a class="a" href="http://www.youtube.com/watch?v=KbF7RMk_2Qo">Video</a></li>
+        <li class="ul__li"><strong>RailsClub 2013</strong> <a class="a" href="http://code.jeremyevans.net/presentations/railsclub2013/index.html?trans=no">Give-and-Go with PostgreSQL and Sequel</a> <a class="a" href="http://code.jeremyevans.net/presentations/railsclub2013/index.html">In Russian</a> <a class="a a--dead" href="https://web.archive.org/web/20170915000000*/http://live.digicast.ru/ru/view/2116">Video</a></li>
```
Comment (fmaclen) [inline www/pages/links.html.erb:56]: The site hosting the video is no more and I couldn't find another link for the video but I decided to link to the archived page anyways to reference there was a video at some point.

File: www/pages/links.html.erb [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        <li class="ul__li"><strong>Lone Star Ruby Conference 2009</strong> <a class="a" href="http://code.jeremyevans.net/presentations/lsrc2009_presentation/sequel-lsrc2009.html">Sequel: SQL in Ruby</a></li>
+        <li class="ul__li"><strong>RubyKaigi 2009</strong> <a class="a" href="http://code.jeremyevans.net/presentations/rk2009_presentation/sequel-rubykaigi2009.html">Sequel: SQL in Ruby</a></li>
+        <li class="ul__li"><strong>LA Ruby Conf 2009</strong> <a class="a" href="http://code.jeremyevans.net/presentations/larc2009_presentation/sequel-larc2009-pres.html">Sequel: The Database Toolkit for Ruby</a> <a class="a" href="https://www.youtube.com/watch?v=0Euba97UEdI">Video</a></li>
+        <li class="ul__li"><strong>MountainWest RubyConf 2009</strong> <a class="a" href="http://code.jeremyevans.net/presentations/mwrc2009_presentation.html">Sequel: The Database Toolkit for Ruby</a> <a class="a" href="https://www.youtube.com/watch?v=yCjtdfv9Z_0">Video</a> <a class="a" href="http://code.jeremyevans.net/presentations/mwrc2009_presentation.txt">Transcript</a></li>
```
Comment (fmaclen) [inline www/pages/links.html.erb:62]: Updated video source to YouTube.

**Comment by fmaclen** [comment]
@jeremyevans I couldn't figure out if there was another script or GitHub Workflow that would run `make__www` and move the files to the folder structure that the `gh_pages` has.

**Comment by jeremyevans** [comment]
This looks great!  I'll merge this on Monday.

**Comment by jeremyevans** [comment]
> @jeremyevans I couldn't figure out if there was another script or GitHub Workflow that would run `make__www` and move the files to the folder structure that [`gh_pages`](https://github.com/jeremyevans/sequel/tree/gh-pages) has.

`rake website website_rdoc` will build the `gh-pages` website in the `www/public` folder.

**Comment by fmaclen** [comment]
Ah, understood, appreciate the clarification!

**Comment by fmaclen** [comment]
One last thing, just added a favicon so the site is easier to spot when you have multiple browser tabs open.

File: www/pages/plugins.html.erb [hunk 7]
```diff
@@ -1,309 +1,1087 @@
...
+<a class="a" href="rdoc-plugins/classes/Sequel/Plugins/ConcurrentEagerLoading.html">concurrent_eager_loading </a>
+<span class="ul__span">Allows you to concurrently eagerly load multiple associations using the async_thread_pool Database extension.</span>
+</li>
+<li class="ul__li ul__li--grid">
+<a class="a" href="rdoc-plugins/classes/Sequel/Plugins/DatasetAssociations.html">dataset_associations </a>
+<span class="ul__span">Adds association methods to datasets that return datasets of associated objects.</span>
+</li>
+<li class="ul__li ul__li--grid">
+<a class="a" href="rdoc-plugins/classes/Sequel/Plugins/DelayAddAssociation.html">delay_add_association </a>
+<span class="ul__span">Delay add_ <i>association</i> calls on new objects until after after the object is saved. </span>
```
Comment (janko) [inline www/pages/plugins.html.erb:78]: There is an extra space between "add_" and "association".

File: www/public/css/application.css [hunk 8]
```diff
@@ -1,238 +1,336 @@
...
-table tbody tr td img { vertical-align: bottom; }
 
-table tbody tr.cmAction {
-	background-color: #FAFAFA;
-	border-top: 1px dotted brown;
-	border-bottom: 1px dotted brown;
-	color: black;
+li.ul__li--grid {
+	display: grid;
+	grid-template-columns: 1fr 3fr;
```
Comment (janko) [inline www/public/css/application.css:302]: There are longer plugin names that exceed the default width of the left-hand column:

Screenshot 2022-04-17 at 10 16 29

**Review by janko — COMMENTED** [review]
Love the new design ❤️ There are couple of small things I noticed, so I thought I'd post them here.

File: www/public/css/application.css (hunk 8 above)
Comment (fmaclen) [reply www/public/css/application.css]: I ended up changing the column ratio from `1fr 3fr` to `1fr 2fr`, that gives it enough room for the current plugins name length (and a few more extra characters too).

If longer much names are expected in the future this component might need to be replaced as each row in the list it's own _little grid_ independent from the other ones.

File: www/pages/index.html.erb [hunk 9]
```diff
@@ -1,40 +1,47 @@
...
+<span class="constant">DB</span> <span class="punct">=</span> <span class="constant">Sequel</span><span class="punct">.</span><span class="ident">sqlite</span>
 
-<span class='comment'># create a dataset from the items table</span>
-<span class='ident'>items</span> <span class='punct'>=</span> <span class='constant'>DB</span><span class='punct'>[</span><span class='symbol'>:items</span><span class='punct'>]</span>
+<span class="comment"># create an items table</span>
+<span class="constant">DB</span><span class="punct">.</span><span class="ident">create_table</span> <span class="symbol">:items</span> <span class="keyword">do</span>
+<span class="ident">primary_key</span> <span class="symbol">:id</span>
+<span class="ident">String</span> <span class="symbol">:name</span><span class="punct">,</span> <span class="symbol">unique:</span> <span class="keyword">true</span><span class="punct">,</span> <span class="symbol">null:</span> <span class="keyword">false</span>
+<span class="ident">Float</span> <span class="symbol">:price</span><span class="punct">,</span> <span class="symbol">null:</span> <span class="keyword">false</span>
+<span class="keyword">end</span>
```
[the remark this answers is not in the document]
Comment (fmaclen) [reply www/pages/index.html.erb]: Good catch, fixed!

**Comment by jeremyevans** [comment]
Squash merged at 624f82e2133bf21e82e0a7e979aa38e5261e2e75

**Pull request closed without merging**